src/pool/srv_pool.c

```diff
@@ -3135,26 +3333,40 @@ ds_pool_connect_handler(crt_rpc_t *rpc, int handler_version)
 	if (rc != 0)
 		D_GOTO(out_map_version, rc);
 
+out_map_version:
+	out->pco_op.po_map_version = ds_pool_get_version(svc->ps_pool);
+
+	/* If meets criteria (not dup, write op, definitive rc, etc.), store result in ps_ops KVS */
+	if ((rc == 0) && !dup_op && fi_fail_noreply)
+		rc = pool_op_save_dup(&tx, svc, rpc, dup_op, handler_version, -DER_MISC, &op_val);
+	else
+		rc = pool_op_save_dup(&tx, svc, rpc, dup_op, handler_version, rc, &op_val);
+	if (rc != 0)
+		goto out_lock;
 	rc = rdb_tx_commit(&tx);
-	if (rc)
-		D_GOTO(out_map_version, rc);
+	if (rc != 0)
+		goto out_lock;
 
-	/** update metric */
-	metrics = svc->ps_pool->sp_metrics[DAOS_POOL_MODULE];
-	d_tm_inc_counter(metrics->connect_total, 1);
+	rc = op_val.ov_rc;
+	if ((rc == 0) && !dup_op) {
+		/** update metric */
+		metrics = svc->ps_pool->sp_metrics[DAOS_POOL_MODULE];
+		d_tm_inc_counter(metrics->connect_total, 1);
+	}
 
-	if (query_bits & DAOS_PO_QUERY_SPACE)
+	if ((rc == 0) && (query_bits & DAOS_PO_QUERY_SPACE))
 		rc = pool_space_query_bcast(rpc->cr_ctx, svc, in->pci_op.pi_hdl, &out->pco_space);
-out_map_version:
-	out->pco_op.po_map_version = ds_pool_get_version(svc->ps_pool);
+
 out_lock:
 	ABT_rwlock_unlock(svc->ps_lock);
 	rdb_tx_end(&tx);
 	if (rc == 0 && transfer_map) {
 		rc = ds_pool_transfer_map_buf(map_buf, map_version, rpc, bulk,
 					      &out->pco_map_buf_size);
-		/** TODO: roll back tx if transfer fails? */
+		/** TODO: roll back tx if transfer fails? Perhaps rdb_tx_discard()? */
```

Comment:
As you know, `rdb_tx_discard` is meant for discarding writes in an uncommitted RDB TX; out-of-date comment, perhaps?